### ISSUE TITLE
Path variables à la express

### ISSUE THREAD

**Issue by enricodeleo** [jkyberneees/fast-gateway#58]
In Express environment is common to express routes like `/user/:userId/group` where `userId` is a variable that can be used across middlewares.

I would find awesome an interface like 

```js
{
    pathRegex: '',
    prefix: '/users/:userId/group',
    target: `http://targetwebsite.com/group?withuser=${userId}`,
}
```

Would it be possible?

**Comment by jkyberneees** [comment]
Hi @enricodeleo, unfortunately this feature is not implemented at the moment. 
However, you can make use of the `onRequest` hook to modify the request URL as desired. 

Can you check that it works for you?

**Comment by enricodeleo** [comment]
Yes I think I'll end up using that hook like I did for response transformations. 

Actually, I think the my proposed interface is near to be available, at least the prefix part is already working. 

In fact, in this scenario

```js
{
    pathRegex: '/:group/:id/products',
    prefix: '/partners',
    prefixRewrite: '/api/v1.0/partners',
    target: API_HOST,
    hooks: {
      rewriteRequestHeaders(req, headers) {
        req.log.debug(req);
        headers['accept-encoding'] = 'identity';
        return headers;
      },
    },
  },
```

I can see `params: { group: 'whatever', id: 'whatever' },` as expected within `req`.

The only part missing is a function that would parse target url as well. It would be super helpful when using fast-gateway as a migration tool for legacy apis (you could use it in order to listen to request urls in a more "standard" way and then piece by piece migrate each endpoint to its own microservice).

**Comment by jkyberneees** [comment]
I understand and I agree with you, let me come up with a flexible approach to update the path or request url without going into the hooks...

Stay tuned

**Comment by enricodeleo** [comment]
Any chances you had some idea on how to address it?

**Comment by jkyberneees** [comment]
Hi @enricodeleo, excuse me for the delay on this.

What do you think about this proposal?
```js
'use strict'

const gateway = require('../index')
const PORT = process.env.PORT || 8080

gateway({
  routes: [{
    pathRegex: '',
    prefix: '/customers/:customerId',
    target: 'http://localhost:3000',
    urlRewrite: ({ params: { customerId } }) => `/users/${customerId}`
  }]
}).start(PORT).then(server => {
  console.log(`API Gateway listening on ${PORT} port!`)
})

const service = require('restana')({})
service
  .get('/users/:id', (req, res) => res.send('Hello ' + req.params.id))
  .start(3000).then(() => console.log('Service listening on 3000 port!'))


```

**Comment by enricodeleo** [comment]
no reason to excuse, you already gave us a great tool! so thank you for the time you spent into this new feature. It does look like a great api !

**Comment by jkyberneees** [comment]
Hi @enricodeleo, the new `urlRewrite` hook is now available in v2.8.0. Thanks for your input!
Have fun ;)
